client/snapshot.go

```diff
@@ -0,0 +1,151 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// errors on reading:
+var (
+	ErrSnapshotUnsupported  = errors.New("unsupported snapshot format")
+	ErrSnapshotSizeMismatch = errors.New("reported size and read size differ")
+	ErrSnapshotHashMismatch = errors.New("hash mismatch")
+)
+
+// errors about not finding stuff (should bubble out as something more user-friendly, probably)
+var (
+	ErrSnapshotNotFound      = errors.New("no snapshot by that ID")
+	ErrSnapshotSnapsNotFound = errors.New("the snapshot with the given ID does not contain the requested snaps")
+)
+
+// A SnapshotOp is used to request an operation on a snapshot
+type SnapshotOp struct {
```

Comment:
I believe the pattern here is snapshotAction for this one.

Reply:
I don't think we have a pattern for this, or rather if we do it's that this struct is not exported (I see `appInstruction` and `actionData` in similar roles for apps and snaps respectively), and not reused from daemon (which is what I'm trying to do here).

Reply:
$ grep 'type.*Action' *
aliases.go:type aliasAction struct {
client.go:type debugAction struct {
interfaces.go:type InterfaceAction struct {
snap_op.go:type multiActionData struct {

$ grep 'type.*Op\>' *    
[niemeyer@nomad ..re/snapd/client]% 

That looks like a pretty clear pattern to me.